lib/msal-common/src/network/NetworkManager.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { INetworkModule } from "./INetworkModule";
+import { ServerTokenRequestParameters } from "../server/ServerTokenRequestParameters";
+import { ICacheStorage } from "../cache/ICacheStorage";
+import { AccessTokenKey } from "../cache/AccessTokenKey";
+import { AccessTokenValue } from '../cache/AccessTokenValue';
+import { Constants } from "../utils/Constants";
+import { ICrypto } from "../crypto/ICrypto";
+import { TokenExchangeParameters } from "../request/TokenExchangeParameters";
+import { ClientInfo } from "../auth/ClientInfo";
+
+export type NetworkResponse<T> = {
+    headers: Map<string, string>;
+    body: T;
+    status: number;
+};
+
+export class NetworkManager {
+    cacheStorage: ICacheStorage;
+    networkClient: INetworkModule;
+    cryptoObj: ICrypto;
+
+    constructor(cacheStorage: ICacheStorage, networkClient: INetworkModule, cryptoObj: ICrypto) {
+        this.cacheStorage = cacheStorage;
+        this.networkClient = networkClient;
+        this.cryptoObj = cryptoObj;
+    }
+
+    async sendPostRequest(tokenEndpoint: string, tokenReqParams: ServerTokenRequestParameters, tokenRequest: TokenExchangeParameters, clientInfo: ClientInfo, clientId: string): Promise<ServerAuthorizationTokenResponse> {
+        const thumbprint = new AccessTokenKey(
+            tokenRequest.authority, 
+            clientId, 
+            tokenRequest.scopes.join(" "), 
```

Comment:
given you will JSON stringify this to act as the cache key, should these string all be normalized with .toLower, and also for scopes, they should be sorted? Or perhaps the AccessTokenKey class should have a method / property for getting or computing the fingerprint (and thus, these details could be handled internally to that class)?

Reply:
These are all valid points. @pkanher617 do you have insight here based on your testing/knowledge so far?